src/ServiceControl.Monitoring/QueueLength/QueueLengthReportHandler.cs

```diff
@@ -1,22 +1,27 @@
 ﻿namespace ServiceControl.Monitoring.QueueLength
 {
     using System.Threading.Tasks;
+    using Infrastructure;
     using NServiceBus;
     using NServiceBus.Metrics;
 
     class QueueLengthReportHandler : IHandleMessages<MetricReport>
     {
+        EndpointRegistry endpointRegistry;
         QueueLengthDataStore queueLengthDataStore;
 
-        public QueueLengthReportHandler(QueueLengthDataStore queueLengthDataStore)
+        public QueueLengthReportHandler(EndpointRegistry endpointRegistry, QueueLengthDataStore queueLengthDataStore)
         {
+            this.endpointRegistry = endpointRegistry;
             this.queueLengthDataStore = queueLengthDataStore;
         }
 
         public Task Handle(MetricReport message, IMessageHandlerContext context)
         {
             var endpointInstanceId = EndpointInstanceId.From(context.MessageHeaders);
 
+            endpointRegistry.Record(endpointInstanceId);
```

Comment:
If we introduced a handler for `IMessage` or `object` we'd be consuming any message. In this case, any misconfigured endpoint sending messages to this queue, would send them into a black hole. Wouldn't it be better to state it in an explicit way, making the subset of accepted messages?